tools/kwiver_tool_runner.cxx

```diff
@@ -0,0 +1,123 @@
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include "kwiver_applet.h"
+#include "applet_context.h"
+
+#include <vital/plugin_loader/plugin_manager.h>
+#include <vital/plugin_loader/plugin_factory.h>
+
+#include <kwiversys/SystemTools.hxx>
+
+#include <cstdlib>
+#include <iostream>
+
+typedef kwiver::vital::implementation_factory_by_name< kwiver::tools::kwiver_applet > applet_factory;
+
+//
+// Global shared context
+//
+kwiver::tools::applet_context G_context;
+
+// ============================================================================
+int main(int argc, char *argv[])
+{
+  kwiver::vital::plugin_manager& vpm = kwiver::vital::plugin_manager::instance();
+  vpm.load_all_plugins();
+
+  // initialize the global context
+  G_context.m_ostream = &std::cout;
+  G_context.m_wtb.set_indent_string( "      " );
+
+  // Test to see if there are any args
+  if ( (argc < 2) || (std::string(argv[1]) == "help") )
+  {
+    // display help message
+    std::cout << "Usage: " << argv[0] << " <applet?  [args]" << std::endl
+              << "<applet> can be one of the following:" << std::endl
+              << "help - prints this message" << std::endl
+      ;
+
+    // Get list of factories for implementations of the applet
+    const auto fact_list = vpm.get_factories( typeid( kwiver::tools::kwiver_applet ).name() );
+
+    // Loop over all factories in the list and display name and description
+    for( auto fact : fact_list )
+    {
+      std::string buf = "-- Not Set --";
+      fact->get_attribute( kwiver::vital::plugin_factory::PLUGIN_NAME, buf );
+      std::cout << "    " << buf << " - ";
+
+      fact->get_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION, buf );
+      std::cout << G_context.m_wtb.wrap_text( buf );
+    }
+
+    exit(0);
+  } // end help code
+
+  // ----------------------------------------------------------------------------
+  try
+  {
+    // Create applet based on the name provided
+    applet_factory app_fact;
+    auto applet = app_fact.create( argv[1] );
+    applet->initialize( &G_context );
+
+    // Edit the command line args
+    // TBD Need to add support for -o <file> flag to direct output to a file.
```

Comment:
If we are going to have options like this at the tool_runner level, maybe they should need to come before the applet name, like `kwiver -o logfile.txt <applet> [applet options]`.

I could change this, but currently I use `-o` to specify output locations to dump config files in the tools in MAP-Tk.

We will probably also want to have default config files installed for each tool and loaded automatically with the option of specifying an additional config file on the command line to override parameters.  I'm not sure if this should be for all tools, or left up to each tool to implement